planner/core/optimizer.go

```diff
@@ -120,6 +121,68 @@ func CheckPrivilege(activeRoles []*auth.RoleIdentity, pm privilege.Manager, vs [
 	return nil
 }
 
+// VisitInfo4PrivCheck generates privilege check infos because privilege check of local temporary tables is different
+// with normal tables. `CREATE` statement needs `CREATE TEMPORARY TABLE` privilege from the database, and subsequent
+// statements do not need any privileges.
```

Comment:
> `CREATE` statement needs `CREATE TEMPORARY TABLE` privilege from the database, and subsequent
// statements do not need any privileges.

Why not check it in the DDL operation?
If you add it as an additional check, it slow down all operations a bit which is unnecessary.

Reply:
I know, but there are too many operations to check, not only DDL but also DML and others.
If I add all these checks in the `planbuilder`, it will be too dirty.

Reply:
Besides, in `planbuilder`, I cannot add `visitInfo` but do not require privilege checks in a graceful way. `visitInfo` and privilege checks are bound together.